src/modules/api.js

```diff
@@ -48,27 +49,46 @@ export default class API {
     } catch {
       console.error('Metamask not found')
     }
-    const state = this.store.getState()
-    const address = getAddress(state)
 
     const eth = this.getEth()
-    vesting = new eth.Contract(vestingAbi, address)
+    const chainId = await eth.getChainId()
+    const TokenAddress = TokenAddressByChainId[chainId]
+
     tokenContracts = {
       [TokenAddress.MANA]: new eth.Contract(manaAbi, TokenAddress.MANA),
       [TokenAddress.DAI]: new eth.Contract(daiAbi, TokenAddress.DAI),
       [TokenAddress.USDT]: new eth.Contract(usdtAbi, TokenAddress.USDT),
       [TokenAddress.USDC]: new eth.Contract(usdcAbi, TokenAddress.USDC),
     }
+
     return this.localWallet
   }
 
   async fetchContract() {
     const state = this.store.getState()
     const address = getAddress(state)
+    const eth = this.getEth()
+
+    let version
+
+    try {
+      vesting = new eth.Contract(periodicTokenVestingAbi, address)
+      await vesting.methods.getIsLinear().call()
+      version = 'v2'
+    } catch (e) {
+      vesting = new eth.Contract(vestingAbi, address)
+      version = 'v1'
+    }
+
+    let tokenContractAddress
+
+    if (version === 'v1') {
```

Comment:
Versions could be extracted to constants to avoid repetition

Reply:
Addressed in https://github.com/decentraland/vesting-dashboard/pull/45/commits/054b1dfffc8d1d2c57a20cef6babadfc8ab2a996